pfcpiface/fteid.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Canonical Ltd.
+
+package pfcpiface
+
+import (
+	"errors"
+	"math"
+	"sync"
+)
+
+const (
+	minValue = 1
+	maxValue = math.MaxUint32
+)
+
+type FTEIDGenerator struct {
+	lock    sync.Mutex
+	offset  uint32
+	usedMap map[uint32]bool
+}
+
+func NewFTEIDGenerator() *FTEIDGenerator {
+	return &FTEIDGenerator{
+		offset:  0,
+		usedMap: make(map[uint32]bool),
+	}
+}
+
+// Allocate and return an id in range [minValue, maxValue]
+func (idGenerator *FTEIDGenerator) Allocate() (uint32, error) {
```

Comment:
Not efficient.. In case we establish few hundred or thousand tunnels then this will become slow. Anything better we can do?

Reply:
This is using the exact same algorithm as was used in the SMF. While we may want to change it for something efficient, I would decouple this performance improvement from this change here which is focused on moving f-teid allocation from the SMF to the UPF. 

Reference:
- https://github.com/omec-project/util/blob/main/idgenerator/idgenerator.go#L40